src/class/macros.rs

```diff
@@ -319,17 +313,13 @@ macro_rules! py_dummy_ternary_self_func {
             use $crate::ObjectProtocol;
 
             let py = $crate::Python::assume_gil_acquired();
-            let _pool = $crate::GILPool::new(py);
-            let slf_ = py.from_borrowed_ptr::<$crate::PyCell<T>>(slf);
-            let arg = py.from_borrowed_ptr::<$crate::PyAny>(arg1);
-            let result = call_mut!(slf_, $f, arg);
-            match result {
-                Ok(_) => {
-                    ffi::Py_INCREF(slf);
-                    slf
-                }
-                Err(e) => e.restore_and_null(py),
-            }
+            $crate::catch_unwind!(py, {
+                let _pool = $crate::GILPool::new(py);
+                let slf_cell = py.from_borrowed_ptr::<$crate::PyCell<T>>(slf);
+                let arg1 = py.from_borrowed_ptr::<$crate::PyAny>(arg1);
+                call_mut!(slf_cell, $f, arg1)?;
```

Comment:
```suggestion
                call_mut!(slf_cell, $f, arg1)?;
                ffi::Py_INCREF(slf);
```
Here we need `INCREF` not to release the object.

Reply:
Thanks, that's fixed it 👍